aws_doc_sdk_examples_tools/doc_gen.py

```diff
@@ -47,6 +49,46 @@ class DocGenMergeWarning(MetadataError):
     pass
 
 
+class ConfigLoader(ABC):
+    @abstractmethod
+    def load(self, filename: str) -> Tuple[Path, Any]:
+        pass
+
+
+class NoneLoader(ConfigLoader):
+    def load(self, filename: str) -> Tuple[Path, Any]:
+        return Path(filename), yaml.safe_load("")
+
+
+class FileLoader(ConfigLoader):
+    def __init__(self, root: Optional[Path] = None):
+        self.config = root or Path(__file__).parent / "config"
+
+    def load(self, filename: str) -> Tuple[Path, Any]:
+        path = self.config / filename
+        print(f"Opening {path}")
```

Comment:
Should we use a logger here so we can control the verbosity?

Reply:
Eventually. Removed for now.